GOAT-Compiler/Code Generation/CNCMachine.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+
+namespace GOAT_Compiler.Code_Generation
+{
+    /// <summary>
+    /// The class that holds information about the CNC machine.
+    /// </summary>
+    public class CNCMachine
+    {
+        private double _currentExtrusion = 0;
+        private double _bedTemp = 0;
+        private double _extruderTemp = 0;
+        private double _fanPower = 0;
+
+        /// <summary>
+        /// The current positon of the extruder.
+        /// </summary>
+        public Vector Position { get; set; } = new Vector(0, 0, 0);
+
+        /// <summary>
+        /// The current extrusion.
+        /// </summary>
+        public double CurrentExtrusion
+        {
+            get
+            {
+                return _currentExtrusion;
+            } 
+            set 
+            {
+                if (value < 0)
+                {
+                    throw new Exception("Extrusion cannot be negative.");
+                }
+                _currentExtrusion = value;
+            } 
+        }
+
+        /// <summary>
+        /// The current extruder rate.
+        /// </summary>
+        public double ExtrusionRate { get; set; } = 0;
+        
+        /// <summary>
+        /// The current bed temperature.
```

Comment:
Plz define which bed? Aka the hot-bed of the printer

Reply:
Renamed to HotBedTemp